netbox/dcim/views.py

```diff
@@ -1908,6 +1911,73 @@ class InterfaceCreateView(generic.ComponentCreateView):
     model_form = forms.InterfaceForm
     template_name = 'dcim/device_component_add.html'
 
+    def post(self, request):
```

Comment:
This is quite a bit to replicate from ComponentCreateView. Can we instead refactor ComponentCreateView to introduce a `on_success()` method or something to that effect? That would allow us to override just that portion of the view logic.

Reply:
I agree, and I didn't really like copying and pasting the method over — I was concerned about introducing an API change to the base class just to implement a slightly different logical change in a subclass. However, let me take a look and see if I can determine a functional way to break up the logic in a more reusable way.

Reply:
2d2719c should resolve this.